Lib/pathlib/_os.py

```diff
@@ -174,40 +174,34 @@ def copyfileobj(source_f, target_f):
 file_metadata_keys = frozenset(file_metadata_keys)
 
 
-def read_file_metadata(path, keys=None, *, follow_symlinks=True):
+def read_file_metadata(path, metadata_keys, *, follow_symlinks=True, dir_entry=None):
     """
     Returns local path metadata as a dict with string keys.
     """
-    if keys is None:
-        keys = file_metadata_keys
-    assert keys.issubset(file_metadata_keys)
-    result = {}
-    for key in keys:
-        if key == 'xattrs':
-            try:
-                result['xattrs'] = [
-                    (attr, os.getxattr(path, attr, follow_symlinks=follow_symlinks))
-                    for attr in os.listxattr(path, follow_symlinks=follow_symlinks)]
-            except OSError as err:
-                if err.errno not in (EPERM, ENOTSUP, ENODATA, EINVAL, EACCES):
-                    raise
-            continue
-        st = os.stat(path, follow_symlinks=follow_symlinks)
-        if key == 'mode':
-            result['mode'] = stat.S_IMODE(st.st_mode)
-        elif key == 'times_ns':
-            result['times_ns'] = st.st_atime_ns, st.st_mtime_ns
-        elif key == 'flags':
-            result['flags'] = st.st_flags
-    return result
+    metadata = {}
+    if 'mode' in metadata_keys or 'times_ns' in metadata_keys or 'flags' in metadata_keys:
+        st = (dir_entry or path).stat(follow_symlinks=follow_symlinks)
```

Comment:
For safety, I would wrap those calls in an `except OSError`.

Reply:
We want to propagate any errors though! Suppressing them doesn't serve users in this case.

Reply:
Oh my bad!

Reply:
To elaborate: we generally only suppress errors from places like `Path.glob()` and `Path.walk()`, where arguably users aren't interested in processing files/directories that are unreadable for whatever reason. But the equation changes when we're doing a write operation - `copy()` or `move()` silently discarding files/metadata won't fly.